rpc.go

```diff
@@ -247,6 +197,35 @@ func (r *rpc) Clear(in *kvv1.Request, _ *kvv1.Response) error {
 	return nil
 }
 
+func getStorage(in *kvv1.Request, span trace.Span, r *rpc, op errors.Op) (kv.Storage, error) {
+	if in.GetStorage() == "" {
+		e := errors.Str("no storage provided")
+		span.RecordError(e)
+		return nil, errors.E(op, e)
+	}
+
+	storage, ok := r.storages[in.GetStorage()]
+
+	if !ok {
+		e := errors.Errorf("no such storage: %s", in.GetStorage())
+		span.RecordError(e)
+		return nil, errors.E(op, e)
+	}
+
+	return storage, nil
+}
+
+func composeKeys(in *kvv1.Request) []string {
```

Comment:
to save allocations, result slice should be passed from the caller by the pointer.

Reply:
YES!